apps/wallet/src/ui/app/zustand/coins/index.ts

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { createSelectors } from '_app/zustand/createSelectors';
+import { set as idbSet } from 'idb-keyval';
+import { create } from 'zustand';
+
+interface CoinsState {
+	pinnedCoinTypes: string[];
+	setPinnedCoinTypes: (key: string, pinnedCoins: string[]) => void;
+}
+
+const coinsStoreBase = create<CoinsState>();
+
+const useCoinsStoreBase = coinsStoreBase((set) => ({
+	pinnedCoinTypes: [],
```

Comment:
I think there's a way to initialize the correct values here as well.

Reply:
This is not trivial since the `get` from `idb-keyval` is an async action. If we want to set it there, we would need to do something like this https://github.com/pmndrs/zustand/discussions/676.
Let me know if this is something we wanna do @Jordan-Mysten

Reply:
Eh fair, either is not really ideal.

Reply:
But either way you need to initialize this value somewhere

Reply:
oh I see it's done in the useEffect in the usage hook, that's fine